app/gui2/src/components/SlideIn.vue

```diff
@@ -0,0 +1,54 @@
+<script setup lang="ts">
+import { computed, shallowRef } from 'vue'
+
+/* Animate the width of a panel, while holding the interior width constant.
+ *
+ * Rendering the content at the target width and clipping prevents distracting and expensive reflow.
+ */
+
+// Target size is passed via CSS variable --slide-in-panel-size.
+const props = defineProps<{
+  show: boolean
+  // Default: from right.
```

Comment:
We may use `withDefaults` https://vuejs.org/guide/typescript/composition-api#props-default-values